x-pack/legacy/plugins/siem/public/pages/overview/alerts_by_category/index.test.tsx

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/* eslint-disable react/display-name */
+
+import euiDarkVars from '@elastic/eui/dist/eui_theme_dark.json';
+import { act } from '@testing-library/react';
+import { mount, ReactWrapper } from 'enzyme';
+import React from 'react';
+import { ThemeProvider } from 'styled-components';
+
+import { useQuery } from '../../../containers/matrix_histogram';
+import { wait } from '../../../lib/helpers';
+import { mockIndexPattern, TestProviders } from '../../../mock';
+
+import { AlertsByCategory } from '.';
+
+jest.mock('../../../lib/kibana');
+
+jest.mock('../../../containers/matrix_histogram', () => {
+  return {
+    useQuery: jest.fn(),
+  };
+});
+
+const theme = () => ({ eui: { ...euiDarkVars, euiSizeL: '24px' }, darkMode: true });
+const from = new Date('2020-03-31T06:00:00.000Z').valueOf();
+const to = new Date('2019-03-31T06:00:00.000Z').valueOf();
+
+describe('Alerts by category', () => {
+  let wrapper: ReactWrapper;
+
+  describe('before loading data', () => {
+    beforeAll(async () => {
+      (useQuery as jest.Mock).mockReturnValue({
+        data: null,
+        loading: false,
+        inspect: false,
+        totalCount: null,
+      });
+
+      await act(async () => {
```

Comment:
Would like to know under what kind of situation we would use `act` to verify a component. I tried to remove this layer (Line.45, 61-64) seems no problem to pass as well.